src/OpenTelemetry/Metrics/MetricAggregators/HistogramMetricAggregator.cs

```diff
@@ -62,11 +121,38 @@ internal HistogramMetricAggregator(string name, string description, string unit,
         public void Update<T>(T value)
             where T : struct
         {
-            // TODO: Implement Histogram!
+            // promote value to be a double
+
+            double val;
+            if (typeof(T) == typeof(long))
+            {
+                val = (long)(object)value;
```

Comment:
double or long?

Reply:
We need to cast the value (which is a long) to a long. Then let compiler convert from long to double.